event_orchestration.go

```diff
@@ -0,0 +1,577 @@
+	"net/http"
+
+	"github.com/google/go-querystring/query"
+)
+
+const (
+	eoPath = "/event_orchestrations"
+)
+
+// Orchestration defines a global orchestration to route events the same source
+// to different services.
+type Orchestration struct {
+	APIObject
+	Name         string                      `json:"name,omitempty"`
+	Description  string                      `json:"description,omitempty"`
+	Team         *APIReference               `json:"team,omitempty"`
+	Integrations []*OrchestrationIntegration `json:"integrations,omitempty"`
+	Routes       int32                       `json:"routes,omitempty"`
+	CreatedAt    string                      `json:"created_at,omitempty"`
+	CreatedBy    *APIReference               `json:"created_by,omitempty"`
+	UpdatedAt    string                      `json:"updated_at,omitempty"`
+	UpdatedBy    *APIReference               `json:"updated_by,omitempty"`
+	Version      string                      `json:"version,omitempty"`
+}
+
+// OrchestrationIntegration is a route into an orchestration.
+type OrchestrationIntegration struct {
+	ID         string                              `json:"id,omitempty"`
+	Parameters *OrchestrationIntegrationParameters `json:"parameters,omitempty"`
+}
+
+type OrchestrationIntegrationParameters struct {
+	RoutingKey string `json:"routing_key,omitempty"`
+	Type       string `json:"type,omitempty"`
+}
+
+// ListOrchestrationsResponse is the data structure returned from calling the ListOrchestrations API endpoint.
+type ListOrchestrationsResponse struct {
+	APIListObject
+	Orchestrations []Orchestration `json:"orchestrations"`
+}
+
+// ListOrchestrationsOptions is the data structure used when calling the ListOrchestrations API endpoint.
+type ListOrchestrationsOptions struct {
+	// Limit is the pagination parameter that limits the number of results per
+	// page. PagerDuty defaults this value to 25 if omitted, and sets an upper
+	// bound of 100.
+	Limit uint `url:"limit,omitempty"`
+
+	// Offset is the pagination parameter that specifies the offset at which to
+	// start pagination results. When trying to request the next page of
+	// results, the new Offset value should be currentOffset + Limit.
+	Offset uint `url:"offset,omitempty"`
+
+	// Total is the pagination parameter to request that the API return the
+	// total count of items in the response. If this field is omitted or set to
+	// false, the total number of results will not be sent back from the PagerDuty API.
+	//
+	// Setting this to true will slow down the API response times, and so it's
+	// recommended to omit it unless you've a specific reason for wanting the
+	// total count of items in the collection.
+	Total bool `url:"total,omitempty"`
+
+	SortBy string `url:"sort_by,omitempty"`
+}
+
+// ListOrchestrations lists all the existing event orchestrations.
+//
+// Deprecated: Use ListOrchestrationsWithContext instead.
+func (c *Client) ListOrchestrations(o ListOrchestrationsOptions) (*ListOrchestrationsResponse, error) {
```

Comment:
Since we are adding new functionality here could we leave out the non-context methods so that we are not adding more deprecated code?

Reply:
I am happy to do that, yes.